### DISCUSSION TITLE
Create git source with a specific path

### DISCUSSION THREAD

**Discussion by julienFlowdesk** [fluxcd/flux2#2446] [Q&A] [answered]
Hello, 

I have been trying to setup a GitOps environment looking like that:
On a Dev kubernetes:

- Flux bootstrap on a repo called ops-k8s using the repo's path "/dev" => this works perfectly, i can write yaml to ops-k8s/dev/pod.yaml and it will be deployed by flux.
- Flux create source on a repo called application-k8s => this is working but I can't provide a path as this repo contains, like the other repo, "/dev" and "/prod". The resulting GitRepository listen on the whole repo.

I am not using Kustomize (I saw that the Kustomize source allows for path).

The only solution I see is to cut my application-k8s in multiple repo for each env but maybe there is another solution ?

PS: there is no notion of tenants or anything like that here.

Thank you

**Comment by stefanprodan** [comment, marked as answer]
A GitRepository object does nothing to your cluster, it's the Flux Kustomization object that contains the path and describes how manifests are applied. Please check out the https://fluxcd.io/docs/get-started/

**Comment by julienFlowdesk** [comment]
hello,

Thank you for the quick answer. 
It was indeed a confusion on my part: I thought that the Kustomization object was somehow only related to a repo containing Kustomize templates.
Creating the correct Kustomization object is indeed answering my need.
Cheers
